lib/src/client.dart

```diff
@@ -1253,124 +1236,125 @@ class Client extends MatrixApi {
   /// [mediaId] The media ID from the `mxc://` URI (the path component)
   ///
   ///
-  /// [allowRemote] Indicates to the server that it should not attempt to fetch the media if it is deemed
-  /// remote. This is to prevent routing loops where the server contacts itself. Defaults to
-  /// true if not provided.
+  /// [allowRemote] Indicates to the server that it should not attempt to fetch the media if
+  /// it is deemed remote. This is to prevent routing loops where the server
+  /// contacts itself.
   ///
-  @override
-  // TODO: remove once we are able to autogen this
-  Future<FileResponse> getContent(String serverName, String mediaId,
-      {bool? allowRemote}) async {
-    String path;
-
-    if (await authenticatedMediaSupported()) {
-      path =
-          '_matrix/client/v1/media/download/${Uri.encodeComponent(serverName)}/${Uri.encodeComponent(mediaId)}';
-    } else {
-      path =
-          '_matrix/media/v3/download/${Uri.encodeComponent(serverName)}/${Uri.encodeComponent(mediaId)}';
-    }
-    final requestUri = Uri(path: path, queryParameters: {
-      if (allowRemote != null && !await authenticatedMediaSupported())
-        // removed with msc3916, so just to be explicit
-        'allow_remote': allowRemote.toString(),
-    });
-    final request = http.Request('GET', baseUri!.resolveUri(requestUri));
-    request.headers['authorization'] = 'Bearer ${bearerToken!}';
-    final response = await httpClient.send(request);
-    final responseBody = await response.stream.toBytes();
-    if (response.statusCode != 200) unexpectedResponse(response, responseBody);
-    return FileResponse(
-        contentType: response.headers['content-type'], data: responseBody);
+  /// Defaults to `true` if not provided.
+  ///
+  /// [timeoutMs] The maximum number of milliseconds that the client is willing to wait to
+  /// start receiving data, in the case that the content has not yet been
+  /// uploaded. The default value is 20000 (20 seconds). The content
+  /// repository SHOULD impose a maximum value for this parameter. The
+  /// content repository MAY respond before the timeout.
+  ///
+  ///
+  /// [allowRedirect] Indicates to the server that it may return a 307 or 308 redirect
+  /// response that points at the relevant media content. When not explicitly
+  /// set to `true` the server must return the media content itself.
+  ///
+  Future<FileResponse> getMediaContent(
```

Comment:
Maybe we shouldn't rename it and regress all upgrading clients to the old behaviour? You can still call the original function using super.getContent.